// @progress/kendo-react-data-tools/dist/npm/drag/DropClue.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_dom_1 = require("react-dom");
var kendo_react_common_1 = require("@progress/kendo-react-common");
/**
 * @hidden
 */
var DropClue = /** @class */ (function (_super) {
    __extends(DropClue, _super);
    function DropClue(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {
            height: 0,
            visible: false,
            left: 0,
            top: 0
        };
        _this.hiddenElementRef = React.createRef();
        return _this;
    }
    DropClue.prototype.render = function () {
        var hiddenElement = this.hiddenElementRef.current;
        var dropClue = this.state.visible && kendo_react_common_1.canUseDOM && react_dom_1.createPortal((React.createElement("div", { className: "k-grouping-dropclue", style: {
                zIndex: 10000,
                display: 'block',
                top: this.state.top + 'px',
                left: this.state.left + 'px',
                height: this.state.height + 'px'
            } })), (hiddenElement && hiddenElement.ownerDocument ?
            hiddenElement.ownerDocument.body :
            document.body));
        return (React.createElement(React.Fragment, null,
            dropClue,
            React.createElement("div", { ref: this.hiddenElementRef, style: { display: 'none' } })));
    };
    return DropClue;
}(React.PureComponent));
exports.DropClue = DropClue;
